File: TextosJapones/src/main/java/org/jisho/textosJapones/model/entities/Estatistica.java
package org.jisho.textosJapones.model.entities;

import java.util.UUID;

public class Estatistica {

    private UUID id;
    private String kanji;
    private String tipo;
    private String leitura;
    private Double quantidade;
    private Float percentual;
    private Double media;
    private Float percentMedia;
    private Integer corSequencial;
    private boolean gerar = false;

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getKanji() {
        return kanji;
    }

    public void setKanji(String kanji) {
        this.kanji = kanji;
    }

    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public String getLeitura() {
        return leitura;
    }

    public void setLeitura(String leitura) {
        this.leitura = leitura;
    }

    public Double getQuantidade() {
        return quantidade;
    }

    public void setQuantidade(Double quantidade) {
        this.quantidade = quantidade;
    }

    public Float getPercentual() {
        return percentual;
    }

    public void setPercentual(Float percentual) {
        this.percentual = percentual;
    }

    public Double getMedia() {
        return media;
    }

    public void setMedia(Double media) {
        this.media = media;
    }

    public Float getPercentMedia() {
        return percentMedia;
    }

    public void setPercentMedia(Float percentMedia) {
        this.percentMedia = percentMedia;
    }

    public Integer getCorSequencial() {
        return corSequencial;
    }

    public void setCorSequencial(Integer corSequencial) {
        this.corSequencial = corSequencial;
    }

    public boolean isGerar() {
        return gerar;
    }

    public void setGerar(boolean gerar) {
        this.gerar = gerar;
    }

    public Estatistica() {
        this.id = null;
        this.kanji = "";
        this.tipo = "";
        this.leitura = "";
        this.quantidade = 0.0d;
        this.percentual = 0.0f;
        this.media = 0.0d;
        this.percentMedia = 0.0f;
        this.corSequencial = 0;
    }

    public Estatistica(String tipo) {
        this.tipo = tipo;
    }

    public Estatistica(UUID id, String kanji, String tipo, String leitura, Double quantidade, Float percentual, Double media,
                       Float percentMedia, Integer corSequencial) {
        this.id = id;
        this.kanji = kanji;
        this.tipo = tipo;
        this.leitura = leitura;
        this.quantidade = quantidade;
        this.percentual = percentual;
        this.media = media;
        this.percentMedia = percentMedia;
        this.corSequencial = corSequencial;
    }

}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/model/entities/processatexto/Estatistica.kt
package br.com.fenix.processatexto.model.entities.processatexto

import br.com.fenix.processatexto.model.entities.EntityBase
import jakarta.persistence.*
import java.util.UUID

@Entity
data class Estatistica(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "ID", nullable = false, unique = true)
    private var id: UUID? = null,
    @Column(length = 10, nullable = true)
    var kanji: String = "",
    @Column(length = 10, nullable = true)
    var tipo: String = "",
    @Column(length = 10, nullable = true)
    var leitura: String = "",
    @Column(nullable = true)
    var quantidade: Double = 0.0,
    @Column(nullable = true)
    var percentual: Float = 0f,
    @Column(nullable = true)
    var media: Double = 0.0,
    @Column(name = "percentual_medio", nullable = true)
    var percentMedia: Float = 0f,
    @Column(name = "cor_sequencial", nullable = true)
    var corSequencial: Int = 0,
    var isGerar : Boolean = false
) : EntityBase<UUID?, Estatistica>() {

    constructor(tipo: String) : this() {
        this.tipo = tipo
    }

    constructor(
        id: UUID, kanji: String, tipo: String, leitura: String, quantidade: Double, percentual: Float, media: Double,
        percentMedia: Float, corSequencial: Int
    ) : this() {
        this.id = id
        this.kanji = kanji
        this.tipo = tipo
        this.leitura = leitura
        this.quantidade = quantidade
        this.percentual = percentual
        this.media = media
        this.percentMedia = percentMedia
        this.corSequencial = corSequencial
    }

    override fun getId(): UUID? = id

    override fun create(id: UUID?): Estatistica = Estatistica(id)

    fun setId(id: UUID?) {
        this.id = id
    }

}